### ISSUE TITLE
Deadlock

### ISSUE THREAD

**Issue by DrAma999** [NordicSemiconductor/IOS-CoreBluetooth-Mock#11]
I've found a potential deadlock while trying to simulate a write without response. If I create a `CBCentralManager` mock that uses the main queue, the code will crash trying to dispatch synchronously on main. `CBMCentralManagerMock.swift` about line 1105 method 
```swift
public func writeValue(_ data: Data,
                           for characteristic: CBMCharacteristic,
                           type: CBMCharacteristicWriteType)
```
Here
```swift
else {
            queue.sync { //<--Here
                guard availableWriteWithoutResponseBuffer > 0 else {
                    return
                }
                availableWriteWithoutResponseBuffer -= 1
                _canSendWriteWithoutResponse = false
            }
```

**Comment by philips77** [comment]
Could you prepare a PR with a fix, please?

**Comment by DrAma999** [comment]
Sure, I'll do in the weekend

**Comment by philips77** [comment]
Closed by #12.
